File: tests/upload-question.py
import requests
import json

question = json.dumps({
    'title': 'Hello World',
    'body': 'I like apples'
})
res = requests.post('http://127.0.0.1:8080/upload-question', data=question, headers={'Content-Type': 'application/json'})
print(res.content)
print(res.status_code)

File: src/routes/uploadQuestion.js
const express = require('express');
const {writeData} = require('../firebase');
const router = express.Router();

router.post('/upload-question', (req, res) => {
  try {
    const {title, body, imageURL, flairs} = req.body;
    const userID = req.auth.user_id;

    const data = {
      userID: userID,
      title: title,
      body: body,
      imgURL: imageURL,
      flairs: flairs ? flairs : [],
      answered: false
    };

    writeData('questions', data, questionID => {
      res.status(200).json({success: `Question successfully uploaded under the ID ${questionID}!`});
    });
  }
  catch (error) {
    res.status(500).json({error: 'Internal server error.'});
  }
})

module.exports = router;